src/routes/plans/[id]/+page.svelte

```diff
@@ -271,6 +276,34 @@
     }
   }
 
+  async function onCallExtension(event: CustomEvent<Extension>) {
+    const { detail } = event;
+    const body = {
+      planId: $planId,
+      selectedActivityDirectiveId: $selectedActivityDirectiveId,
+      simulationDatasetId: $simulationDatasetId,
+      url: detail.url,
+    };
+    const headers: HeadersInit = {
```

Comment:
Does it make sense to swap this with the `reqExtension` function and have the server build the header/payload? My thinking was that this logic could potentially be called from different places in the application (unlikely but possible) so having it go through a function would make things less verbose.

BONUS: Is there any way you can make it so that the server and the UI can call the same `reqExtension` function? We have instances where the server calls `reqHasura` to pre-populate the page via the `effects` file.

Reply:
Done!